# Análise Exploratória do Mercado Imobiliário da Cidade de São Paulo


## 1.0 - Contextualização

O mercado imobiliário da cidade de São Paulo é um dos maiores e mais diversificados do Brasil, reunindo imóveis com diferentes características, localizações e faixas de preço. Essa diversidade torna o processo de avaliação e comparação de imóveis um desafio para compradores, vendedores, investidores e empresas do setor.

Com o crescimento da disponibilidade de dados, técnicas de Análise Exploratória de Dados (Exploratory Data Analysis — EDA) tornaram-se uma ferramenta importante para compreender o comportamento do mercado imobiliário. A partir da análise dos dados, é possível identificar padrões, tendências e relações entre características dos imóveis e seus respectivos valores, fornecendo informações que apoiam a tomada de decisão.

Neste projeto será realizada uma análise exploratória de uma base de dados pública contendo anúncios de imóveis da cidade de São Paulo. O foco será compreender a estrutura dos dados, prepará-los para análise e investigar como fatores como localização, área útil, número de dormitórios, banheiros, vagas de garagem, condomínio e IPTU podem estar relacionados ao preço dos imóveis.

Além de gerar insights sobre o mercado imobiliário, este projeto tem como objetivo demonstrar a aplicação de boas práticas em um fluxo de trabalho de Análise de Dados, incluindo inspeção, preparação, exploração e comunicação dos resultados, utilizando Python e suas principais bibliotecas para ciência de dados.

## 2.0 - Objetivo do Projeto

### 2.1 - Objetivo Central

Realizar uma Análise Exploratória de Dados (Exploratory Data Analysis — EDA) sobre uma base de imóveis da cidade de São Paulo, com o objetivo de compreender as características do mercado imobiliário, identificar fatores associados ao preço dos imóveis e gerar insights que possam apoiar a tomada de decisão de compradores, vendedores, investidores e profissionais do setor.

### 2.2 - Objetivos específicos

Para alcançar o objetivo geral deste projeto, serão desenvolvidas as seguintes atividades:

    → Compreender a estrutura e o conteúdo da base de dados.
    → Inspecionar a qualidade dos dados, identificando possíveis inconsistências, valores ausentes e registros duplicados.
    → Preparar os dados para análise por meio da correção de tipos de dados, tratamento de valores ausentes, padronização e demais transformações necessárias.
    → Realizar análises estatísticas descritivas para compreender o comportamento das variáveis.
    → Explorar a distribuição das principais variáveis por meio de visualizações gráficas.
    → Investigar a relação entre as características dos imóveis e seus respectivos preços.
    → Identificar padrões, tendências e possíveis outliers presentes na base de dados.
    → Responder às perguntas de negócio definidas para o projeto utilizando evidências obtidas na análise.
    → Comunicar os principais insights de forma clara e objetiva, apoiando a tomada de decisão.
    → Desenvolver um projeto estruturado, reproduzível e documentado, seguindo boas práticas de Análise de Dados.


## 3.0 - Perguntas de Negócio

Este projeto busca responder às seguintes perguntas de negócio:

    → Como os preços dos imóveis estão distribuídos na cidade de São Paulo?
    → Quais bairros apresentam os maiores e os menores preços médios de venda?
    → Existe relação entre a área útil do imóvel e o seu preço de venda?
    → Como a quantidade de dormitórios, banheiros, suítes e vagas de garagem influencia o preço dos imóveis?
    → O valor do condomínio apresenta relação com o preço de venda dos imóveis?
    → Quais tipos de imóveis possuem os maiores valores médios de venda?
    → Existem bairros que oferecem melhor custo por metro quadrado em comparação com outros?
    → Há imóveis com valores significativamente acima ou abaixo do padrão da base de dados (outliers)?
    → Quais características apresentam maior associação com o preço dos imóveis?
    → Quais insights podem ser extraídos para apoiar compradores, investidores e profissionais do mercado imobiliário na tomada de decisão?


## 4.0 - Metodologia

Este projeto será desenvolvido seguindo uma adaptação da metodologia CRISP-DM (Cross-Industry Standard Process for Data Mining), amplamente utilizada em projetos de Ciência e Análise de Dados.

O processo está organizado nas seguintes etapas:

    → Compreensão do Negócio (Business Understanding): definição do contexto, dos objetivos e das perguntas de negócio que orientarão toda a análise.
    → Compreensão dos Dados (Data Understanding): carregamento da base de dados, inspeção inicial da estrutura, identificação dos tipos de dados, verificação de valores ausentes, registros duplicados e outras características relevantes.
    → Preparação dos Dados (Data Preparation): tratamento de inconsistências, conversão de tipos de dados, padronização de informações, tratamento de valores ausentes, remoção de duplicidades e demais transformações necessárias para garantir a qualidade da base.
    → Análise Exploratória dos Dados (Exploratory Data Analysis – EDA): aplicação de técnicas estatísticas e visualizações para compreender o comportamento das variáveis, identificar padrões, investigar relações entre os atributos e responder às perguntas de negócio definidas no início do projeto.
    → Conclusão: apresentação dos principais insights obtidos, resposta às perguntas de negócio, discussão das limitações da análise e recomendações para trabalhos futuros.

Todo o desenvolvimento será realizado em ambiente Jupyter Notebook, utilizando Python e bibliotecas especializadas para manipulação, tratamento, análise e visualização de dados, seguindo boas práticas de organização, documentação e reprodutibilidade.


## 5.0 - Ferramentas utilizadas

O desenvolvimento deste projeto será realizado utilizando as seguintes ferramentas e bibliotecas:

    → Python
    → Jupyter Notebook
    → Pandas
    → Numpy
    → Matplotlib


## 6.0 - Base de dados

Este projeto utiliza a base de dados House Prices of Sao Paulo City, disponibilizada publicamente na plataforma Kaggle. O conjunto reúne informações sobre imóveis anunciados para venda na cidade de São Paulo, incluindo variáveis como preço, área, número de dormitórios, banheiros, vagas de garagem, condomínio, IPTU e localização.

Link da base de dados: https://www.kaggle.com/datasets/ex0ticone/house-prices-of-sao-paulo-city

### 6.1 - Dicionário dos Dados

A tabela abaixo descreve as variáveis presentes na base de dados utilizada neste projeto.

| Coluna | Tipo esperado | Descrição |
| :------ | :------------ | :-------- |
| `logradouro` | Texto | Nome da rua, avenida, alameda ou via onde o imóvel está localizado. |
| `numero` | Numérico | Número do imóvel. Pode conter valores ausentes quando não informado. |
| `bairro` | Texto | Bairro onde o imóvel está localizado. |
| `cep` | Texto | Código de Endereçamento Postal (CEP). |
| `cidade` | Texto | Município onde o imóvel está localizado. |
| `tipo_imovel` | Texto | Categoria do imóvel (apartamento, casa, cobertura, sobrado, entre outros). |
| `area_util` | Numérico | Área útil do imóvel em metros quadrados (m²). |
| `banheiros` | Numérico | Quantidade de banheiros do imóvel. |
| `suites` | Numérico | Quantidade de suítes do imóvel. |
| `quartos` | Numérico | Quantidade de dormitórios (quartos). |
| `vagas_garagem` | Numérico | Número de vagas de garagem disponíveis. |
| `anuncio_criado` | Data | Data de publicação do anúncio. |
| `tipo_anuncio` | Texto | Modalidade do anúncio, indicando se o imóvel está disponível para venda ou aluguel. |
| `preco_venda` | Numérico | Valor anunciado para venda do imóvel, em reais. |
| `taxa_condominio` | Numérico | Valor mensal da taxa de condomínio, em reais. |
| `periodicidade` | Texto | Periodicidade dos valores monetários apresentados no anúncio. |
| `preco_aluguel` | Numérico | Valor do aluguel do imóvel, em reais, quando disponível. |
| `iptu_ano` | Numérico | Valor anual do IPTU (Imposto Predial e Territorial Urbano), em reais. |

## 7.0 - Compreensão dos dados (Data Undestanding)

### 7.1 - Importação de bibliotecas

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from pathlib import Path

### 7.2 - Definições

In [2]:
DATA_DIR = Path('../data/raw')
DATA_FILE = DATA_DIR / 'housing_sp_city.csv'

### 7.3 - Carregando base de dados

In [3]:
df_raw = pd.read_csv(DATA_FILE, encoding='latin-1')
df = df_raw.copy()

### 7.4 - Visualização da base de dados

Antes de iniciar qualquer inspeção técnica, é importante realizar uma visualização inicial da base de dados.

Essa etapa permite verificar se o arquivo foi carregado corretamente, conhecer a estrutura das colunas, observar exemplos de registros e identificar possíveis inconsistências visíveis, como valores ausentes, caracteres inesperados ou problemas de formatação.

Neste momento, nenhuma alteração será realizada nos dados. O objetivo é apenas compreender sua organização e obter uma visão geral do conjunto de dados.

In [4]:
df.head()

,logradouro,numero,bairro,cep,cidade,tipo_imovel,area_util,banheiros,suites,quartos,vagas_garagem,anuncio_criado,tipo_anuncio,preco_venda,taxa_condominio,periodicidade,preco_aluguel,iptu_ano
0,Rua Juvenal Galeno,53,Jardim da Saúde,4290030.0,São Paulo,Casa de dois andares,388.0,3.0,1.0,4.0,6.0,2017-02-07,Venda,700000,NaN,NaN,NaN,NaN
1,Rua Juruaba,16,Vila Santa Teresa (Zona Sul),4187320.0,São Paulo,Casa,129.0,2.0,1.0,3.0,2.0,2016-03-21,Venda,336000,NaN,NaN,NaN,NaN
2,Avenida Paulista,402,Bela Vista,1311000.0,São Paulo,Comercial,396.0,4.0,0.0,0.0,5.0,2018-12-18,Locação,24929,4900.0,MONTHLY,29829.0,4040.0
3,Rua Alvorada,1190,Vila Olímpia,4550004.0,São Paulo,Apartamento,80.0,2.0,1.0,3.0,2.0,2018-10-26,Venda,739643,686.0,NaN,NaN,1610.0
4,Rua Curitiba,380,Paraíso,4005030.0,São Paulo,Apartamento,3322.0,5.0,4.0,4.0,5.0,2018-12-14,Venda,7520099,6230.0,NaN,NaN,18900.0


In [5]:
num_linhas, num_colunas = df.shape

print(f'Número de registros: {num_linhas}')
print(f'Número de variáveis: {num_colunas}')

Número de registros: 133964
Número de variáveis: 18


### 7.5 - Tipos de dados

Nesta etapa será realizada a inspeção dos tipos de dados de cada variável presente na base.

A identificação correta dos tipos de dados é fundamental para garantir que cada variável seja tratada adequadamente durante a etapa de preparação dos dados. Além disso, essa verificação permite identificar possíveis inconsistências, como datas armazenadas como texto, valores numéricos importados como caracteres ou variáveis categóricas que poderão ser otimizadas posteriormente.

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 133964 entries, 0 to 133963
Data columns (total 18 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   logradouro       126962 non-null  str    
 1   numero           89823 non-null   str    
 2   bairro           133943 non-null  str    
 3   cep              128056 non-null  float64
 4   cidade           133964 non-null  str    
 5   tipo_imovel      133964 non-null  str    
 6   area_util        130868 non-null  float64
 7   banheiros        133051 non-null  float64
 8   suites           120347 non-null  float64
 9   quartos          130945 non-null  float64
 10  vagas_garagem    129539 non-null  float64
 11  anuncio_criado   133964 non-null  str    
 12  tipo_anuncio     133964 non-null  str    
 13  preco_venda      133964 non-null  int64  
 14  taxa_condominio  117127 non-null  float64
 15  periodicidade    29051 non-null   str    
 16  preco_aluguel    28714 non-null   float64
 17  ip

### 7.6 - Verificação de valores ausentes

Valores ausentes são comuns em bases de dados reais e podem ocorrer por diferentes motivos, como falhas na coleta, informações não disponibilizadas ou características que não se aplicam a determinados registros.

Nesta etapa será realizada a identificação da quantidade de valores ausentes em cada variável da base de dados.

O objetivo é compreender a extensão desse problema e subsidiar as decisões que serão tomadas durante a etapa de preparação dos dados. Neste momento, nenhum tratamento será realizado.

Trataremos posteriormente os valores ausentes obedecendo uma regra geral da análise de dados:

    → menos de 5% → geralmente é possível tratar sem grande impacto;
    → entre 5% e 30% → exige uma análise mais cuidadosa;
    → acima de 30% → pode indicar que a variável merece um tratamento específico ou até mesmo ser descartada, dependendo do contexto.

In [7]:
valores_ausentes = (pd.DataFrame({ 'Valores Ausentes': df.isna().sum(), 
                                   'Percentual (%)' : (df.isna().mean() * 100).round(2)})
                    .sort_values(by='Percentual (%)', ascending=False))

valores_ausentes

,Valores Ausentes,Percentual (%)
preco_aluguel,105250,78.57
periodicidade,104913,78.31
numero,44141,32.95
iptu_ano,19352,14.45
taxa_condominio,16837,12.57
suites,13617,10.16
logradouro,7002,5.23
cep,5908,4.41
vagas_garagem,4425,3.30
area_util,3096,2.31


### 7.7 - Verificação de Registros duplicados

Registros duplicados podem surgir por falhas na coleta, integração de diferentes fontes de dados ou erros durante o processo de armazenamento.

A presença de duplicidades pode comprometer a qualidade das análises, influenciando estatísticas descritivas, distribuições e modelos analíticos.

Nesta etapa será verificada a existência de registros completamente duplicados na base de dados. Caso sejam identificados, sua remoção será avaliada durante a etapa de preparação dos dados.

In [8]:
duplicados = df.duplicated().sum()

print(f'Registros duplicados: {duplicados}')

Registros duplicados: 1271


> A inspeção dos registros duplicados indicou que a base possui **1271** registros completamente duplicados. A decisão sobre a remoção desses registros será tomada na etapa de preparação dos dados, considerando o impacto dessa operação na análise.